### PR TITLE
Proposal: New repository structure

### PR DESCRIPTION
**Pull request by rpitonak**
This PR changing the repository structure.
- every version directory of service now contains directories for every distribution (Fedora, CentOS, RHEL).
- There is one set of the tests for every version.
- Every image has "root" directory, which contains help file   

Open question:
  - where will be common files for every distribution (or version?) placed?

Your feedback is welcome!

### CODE REVIEW

**Comment by TomasTomecek** [comment]
fuf, this is massive: might be even worth talking about this in person

File: Makefile [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
 
-IMAGE_NAME = image
-MODULEMDURL=file://your_module_name.yaml
+VERSION ?= x.y
 
 default: run
 
-build: build-x.y build-x.z
+build:
+	cd $(VERSION)/$(TARGET) && make build VERSION=$(VERSION)
```
Comment (TomasTomecek) [inline Makefile:8]: so `build` will build only a single image? could be interesting to build all of them

File: x.y/centos/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM centos:7
+
+LABEL MAINTAINER ...
```
Comment (TomasTomecek) [inline x.y/centos/Dockerfile:3]: can we have here all required labels mentioned -- ideally in a form so the dockerfile builds?

File: x.y/rhel/examples/run_in_cluster.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+You can use this image to start clustered version of `$service`.
```
Comment (TomasTomecek) [inline x.y/rhel/examples/run_in_cluster.md]: I would get rid of this file since we have help files.

File: README.md [hunk 4]
```diff
@@ -14,20 +14,21 @@ There are two templates within this repository on two different branches:
 
  * `LICENSE` — pick the right license
  * `Makefile` — so it's easy to build, run, test
- * `openshift-template.yml` - general OpenShift template. A postfix template https://github.com/container-images/postfix/blob/master/openshift-template.yml 
+ * `openshift-template.yml` - general OpenShift template. A postfix template https://github.com/container-images/postfix/blob/master/openshift-template.yml
   * update it. It includes several commands what to update
  * `README.md` — global documentation for the whole service
```
Comment (TomasTomecek) [inline README.md]: We should make sure that people are pointed to help files so we don't duplicate.

File: README.md [hunk 5]
```diff
@@ -14,20 +14,21 @@ There are two templates within this repository on two different branches:
...
 ### Image specific files and directories
 
  * `Dockerfile` — default Dockerfile, should be based on Fedora now (will likely change in future)
- * `Dockerfile.$distro` — Dockerfile for other distro, should be also built by Makefile
+ * `Makefile` - so it's easy to build, run, test
  * `examples/` — directory with examples that demonstrate the image functionality:
    * these examples should be ideally present in a form of documentation
    * code samples are welcome too
  * `files/` — this is the place where you should store configuration, helper scripts, etc.
  * `README.md` — version specific documentation
```
Comment (TomasTomecek) [inline README.md]: Similar as above: make sure we link to help file and let's make the readme short.

**Review by TomasTomecek — CHANGES_REQUESTED** [review]
I still prefer we talk about this in person since this is so big.

Edit: ideally I would love to hook CI (out of scope of this PR) so we can be sure that our template can built and tests are passing. Yes, even on template.

File: Makefile (hunk 1 above)
Comment (TomasTomecek) [reply Makefile:8]: We just agreed we would only build the latest version -- we can always improve this later.

**Comment by rpitonak** [comment]
Fixed, can you check it again? @TomasTomecek

**Comment by TomasTomecek** [comment]
Since this is so big, let's merge and address potential issues individually.

**Pull request merged**